### PR TITLE
Introduced verbose feedback on Init in case the app

### PR DESCRIPTION
**Pull request by ruseinov**
was already initialised effectively preventing
a user to do that.
Genesis time is now being updated to init time.
Force flag has been introduced to override re-init
protection.
A message is now being printed if the init was a success.

This still needs tests to be updated.

### CODE REVIEW

**Comment by ruseinov** [comment]
closes https://github.com/confio/weave/issues/46

**Comment by ruseinov** [comment]
Ok, I think this is done.

File: commands/server/init.go [hunk 1]
```diff
@@ -17,9 +18,13 @@ import (
 const (
 	// AppStateKey is the key in the json json where all info
 	// on initializing the app can be found
-	AppStateKey   = "app_state"
-	flagIndexAll  = "all"
-	flagIndexTags = "tags"
+	AppStateKey             = "app_state"
+	DirConfig               = "config"
+	GenesisTimeKey          = "genesis_time"
```
Comment (ethanfrey) [inline commands/server/init.go]: Nice add error message and flags as constants

File: commands/server/init.go [hunk 2]
```diff
@@ -92,7 +103,16 @@ func addGenesisOptions(filename string, options json.RawMessage) error {
 		return err
 	}
 
+	v, ok := doc[AppStateKey]
```
Comment (ethanfrey) [inline commands/server/init.go]: good idea with the force flag, and not just prevent the app_state override always

File: examples/mycoind/commands/init_test.go [hunk 3]
```diff
@@ -40,7 +41,16 @@ func TestInit(t *testing.T) {
...
 	assert.NotEmpty(t, doc[server.AppStateKey])
+
+	gTime := time.Time{}
+	err = gTime.UnmarshalJSON(doc[server.GenesisTimeKey])
+	require.NoError(t, err)
+	assert.True(t, gTime.After(time.Now().Add(-10*time.Minute)))
+
 	assert.Contains(t, string(doc[server.AppStateKey]), `"ticker": "ETH"`)
+
+	err = server.InitCmd(app.GenInitOptions, logger, home, args)
```
Comment (ethanfrey) [inline examples/mycoind/commands/init_test.go]: okay, you test rejecting the second init :+1:
but not with the force flag.... minor point

File: commands/server/init.go [hunk 4]
```diff
@@ -17,9 +18,13 @@ import (
...
 	// AppStateKey is the key in the json json where all info
 	// on initializing the app can be found
-	AppStateKey   = "app_state"
-	flagIndexAll  = "all"
-	flagIndexTags = "tags"
+	AppStateKey             = "app_state"
+	DirConfig               = "config"
+	GenesisTimeKey          = "genesis_time"
+	ErrorAlreadyInitialised = "the application has already been initialised, use %s flag to override"
+	FlagForce               = "f"
```
Comment (ethanfrey) [inline commands/server/init.go]: You don't think this flag should be longer? like `-force`?

I guess `rm -f` is the same.... hmm....

**Comment by ruseinov** [comment]
@ethanfray I didn’t test flags cause that requires abstracting them. Otherwise you need to actually pass flags to command line to test:) we’ll need to do it later, but I felt like testing force is not important.

**Pull request merged**